File: field-force-java/src/Board/Wall.java
package Board;

import AI.AI;
import Game.GameConstants;

public class Wall extends MapObject {
    private int hp = GameConstants.WALL_HP;

    public Wall(char id, int x, int y){
        super(id, x, y);
    }

    public Wall(char id, int x, int y, int hp){
        super(id, x, y);
        this.hp = hp;
    }

    public void takeDamage(int damage)
    {
        hp -= damage;
        if (hp <= 0)
        {
            destroy();
        }
    }

    public int getHp() {
        return hp;
    }

    public void setHp(int hp) {
        this.hp = hp;
    }
}

File: field-force-python/board/wall.py
from board.mapobject import MapObject


class Wall(MapObject):
    def __init__(self, identifier, x, y, hp):
        super().__init__(identifier, x, y)
        self.hp = hp

    def take_damage(self, damage, gamestate):
        self.hp -= damage
        if self.hp <= 0:
            self.destroy(gamestate)
